module/solutions/helper.js

```diff
@@ -1470,6 +1519,173 @@ module.exports = class SolutionsHelper {
     })
    }
 
+   /**
+   * Get link by solution id
+   * @method
+   * @name getLink
+   * @param {String} solutionId - solution Id.
+   * @returns {Object} - Details of the solution.
+   */
+
+   static getLink( solutionId, appName ) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        let solutionData = await this.solutionDocuments({
+                                _id : solutionId,
+                                isReusable : false,
+                                isAPrivateProgram: false
+                                },[
+                                    "link","type","author"
+                              ]);
+
+        if(!Array.isArray(solutionData) || solutionData.length < 1) {
+            return resolve({
+                message: constants.apiResponses.SOLUTION_NOT_FOUND,
+                result: {}
+            });
+        }
+
+        let appDetails = await appsHelper.getDetails(appName);
+
+        if(!appDetails || !appDetails.data){
+          throw new Error(constants.apiResponses.APP_DETAILS_NOT_FOUND);
+        }
+
+        let link;
+
+        if(!solutionData[0].link){
+
+          let updateLink = await gen.utils.md5Hash(solutionData[0]._id + "###" + solutionData[0].author);
+
+          let updateSolution = await this.updateSolutionDocument
```

Comment:
@priyanka-TL  findOneAndUpdate will help right for existing solution doc if its present in db ??